.github/workflows/test_main.yaml

```diff
@@ -0,0 +1,62 @@
+---
+name: Test Main
+on:
+  push:
+    branches:
+      - "*"
+  pull_request:
+    branches:
+      - "*"
+jobs:
+  test:
+    name: Test
+    timeout-minutes: 30
+    strategy:
+      matrix:
+        os: ["ubuntu-latest"]
+        redis: ["5.0"]
+        ruby: ["2.7", "2.6", "2.5", "2.4"]
+        driver: ["ruby", "hiredis", "synchrony"]
+    runs-on: ${{ matrix.os }}
+    env:
+      VERBOSE: true
+      TIMEOUT: 30
+      LOW_TIMEOUT: 0.01
+      DRIVER: ${{ matrix.driver }}
+      REDIS_BRANCH: ${{ matrix.redis }}
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Print environment variables
+        run: |
+          echo "TIMEOUT=${TIMEOUT}"
+          echo "LOW_TIMEOUT=${LOW_TIMEOUT}"
+          echo "DRIVER=${DRIVER}"
+          echo "REDIS_BRANCH=${REDIS_BRANCH}"
+      - name: Set up Ruby
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+      - name: Cache dependent gems
+        uses: actions/cache@v1
+        with:
+          path: .bundle
+          key: "local-bundle-ruby-${{ matrix.ruby }}-on-${{ matrix.os }}-0001"
+      - name: Set up Gems
+        run: |
+          gem update --system --no-document
+          gem install bundler --no-document
+          bundle install --jobs 4 --retry 3 --path=.bundle
+      - name: Cache local temporary directory
+        uses: actions/cache@v1
+        with:
+          path: tmp
+          key: "local-tmp-redis-${{ matrix.redis }}-on-${{ matrix.os }}-0001"
```

Comment:
We can use hash value if we add `Gemfile.lock` file to Git. Is there a policy that we don't control the lock file in the repository?
https://github.com/actions/cache/blob/master/examples.md#ruby---bundler